integration/tykoas_test.go

```diff
@@ -0,0 +1,221 @@
+		tykCtx context.Context
+	)
+
+	f := features.New("Test OAS Delete").
+		Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			// Obtain Environment configuration to be able to connect Tyk.
+			tykEnv, err := generateEnvConfig(ctx, c)
+			eval.NoErr(err)
+
+			res := semver.Compare(tykEnv.TykVersion, "v4.1")
+			if res < 0 {
+				t.Skip("OAS support is added in Tyk in v4.1")
+			}
+
+			tykCtx = tykClient.SetContext(context.Background(), tykClient.Context{
+				Env: tykEnv,
+				Log: log.NullLogger{},
+			})
+
+			var ok bool
+
+			testNS, ok = ctx.Value(ctxNSKey).(string)
+			eval.True(ok)
+
+			tykOAS, _, err = createTestOASApi(ctx, testNS, c)
+			eval.NoErr(err)
+
+			err = waitForTykResourceCreation(c, tykOAS)
+			eval.NoErr(err)
+
+			return ctx
+		}).Assess("Deleting Tyk OAS CRD deletes Tyk OAS API from Tyk",
+		func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			apiID := tykOAS.Status.ID
+
+			err := c.Client().Resources(testNS).Delete(ctx, tykOAS)
+			eval.NoErr(err)
+
+			err = wait.For(conditions.New(c.Client().Resources(testNS)).ResourceDeleted(tykOAS),
+				wait.WithInterval(defaultWaitInterval), wait.WithTimeout(defaultWaitTimeout))
+			eval.NoErr(err)
+
+			exists := klient.Universal.TykOAS().Exists(tykCtx, apiID)
+			eval.True(!exists)
+
+			return ctx
+		}).Feature()
+
+	testenv.Test(t, f)
+}
+
+func TestOASUpdate(t *testing.T) {
+	var (
+		testNS string
+		tykOAS *v1alpha1.TykOAS
+		tykCtx context.Context
+		cm     *v1.ConfigMap
+		tykEnv environment.Env
+	)
+
+	eval := is.New(t)
+
+	f := features.New("Test OAS Update").
+		Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			var err error
+
+			// Obtain Environment configuration to be able to connect Tyk.
+			tykEnv, err = generateEnvConfig(ctx, c)
+			eval.NoErr(err)
+
+			res := semver.Compare(tykEnv.TykVersion, "v4.1")
+			if res < 0 {
+				t.Skip("OAS support is added in Tyk in v4.1")
+			}
+
+			tykCtx = tykClient.SetContext(context.Background(), tykClient.Context{
+				Env: tykEnv,
+				Log: log.NullLogger{},
+			})
+
+			var ok bool
+
+			testNS, ok = ctx.Value(ctxNSKey).(string)
+			eval.True(ok)
+
+			tykOAS, cm, err = createTestOASApi(ctx, testNS, c)
+			eval.NoErr(err)
+
+			err = waitForTykResourceCreation(c, tykOAS)
+			eval.NoErr(err)
+
+			return ctx
+		}).Assess("Updating Configmap updates Tyk OAS API",
+		func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			newListenPath := "/petstore-update"
+			quotedListenPath := fmt.Sprintf("\"%s\"", newListenPath)
+			oasDoc := cm.Data[testOASConfKeyName]
+
+			result, err := jsonparser.Set([]byte(oasDoc), []byte(quotedListenPath),
+				controllers.TykOASExtenstionStr, "server", "listenPath", "value")
+			eval.NoErr(err)
+
+			cm.Data[testOASConfKeyName] = string(result)
+
+			err = c.Client().Resources(testNS).Update(ctx, cm)
+			eval.NoErr(err)
+
+			// Create ApiDefinition Reconciler.
```

Comment:
I guess, instead of `ApiDefinition`, it should be `TykOASReconciler`
```suggestion
			// Create TykOASReconciler.
```